expr/functions/asPercent/function.go

```diff
@@ -46,6 +46,9 @@ func (f *asPercent) Do(ctx context.Context, e parser.Expr, from, until int64, va
 	var denominators []*types.MetricData
 
 	var results []*types.MetricData
+	if len(arg) == 0 {
```

Comment:
Does it maybe makes sense to move after `err != nil` check together with `var results []*types.MetricData` line to reduce memory allocations?